### PR TITLE
Revisit import/export logic, allow running only one locale

### PR DESCRIPTION
**Pull request by flodolo**
The idea is:
* Allow running both import and export on a single locale, adding the `locale` parameter. This is useful for local testing, but also for automation (i.e. only export en-US instead of all locales).
* Importing all locales, i.e. all folders (minus `templates`)  available in the l10n repository. The import currently uses `shipping_locales.txt`,  but I believe the agreement with @delphine was to import and ship any locale with some localization. This also required manually adding brand new locales ([example](https://github.com/mozilla-mobile/firefox-ios/issues/11391)).
* Exporting all locales. For some reason, this function had yet another hard-coded list, which obviously leads to problem since it gets out of sync.

### CODE REVIEW

File: Sources/LocalizationTools/main.swift [hunk 1]
```diff
@@ -38,16 +41,27 @@ struct LocalizationTools: ParsableCommand {
...
         
-        let shippingLocales = URL(fileURLWithPath: projectPath).deletingLastPathComponent().appendingPathComponent("shipping_locales.txt")
-        let locales = try! String(contentsOf: shippingLocales).components(separatedBy: .newlines).filter { !$0.isEmpty }
-
+        var locales: [String]
 
+        if localeCode != nil {
+            locales = [ localeCode! ]
+        } else {
+            let directoryContent = try FileManager.default.contentsOfDirectory(at: URL(fileURLWithPath: l10nProjectPath), includingPropertiesForKeys: nil, options: .skipsHiddenFiles)
```
Comment (flodolo) [inline Sources/LocalizationTools/main.swift:49]: I don't know swift, but this seems incredibly clunky. Hopefully an actual developer has a better idea on how to get the list of non hidden folders in l10nProjectPath :-(

**Comment by flodolo** [comment]
@electricRGB @isabelrios 
More interesting discoveries, since I can import and export a single locale with this change.

The export function had a list of locales, which included `en` (not `en-US`). This exports touches the file in `en-US/firefox-ios.xliff`, with `source-language="en" target-language="en"`.

If you export the `en-US` locale, the `target-language` is `en-US`.

If you import the `en-US` XLIFF file, you'll get a bunch of errors, and nothing is imported. And that's the locale currently defined in `shipping_locales.txt`, used by the import function…

```
xcodebuild: error: contents.json says the source locale is en and the target locale is en-US, but these files differ:
XLIFF has target-language="en" for Shared/Supporting Files/en.lproj/JumpBackIn.strings
XLIFF has target-language="en" for Shared/en.lproj/LoginManager.strings
XLIFF has target-language="en" for Shared/en.lproj/HistoryPanel.strings
XLIFF has target-language="en" for Shared/en.lproj/ClearPrivateData.strings
XLIFF has target-language="en" for Shared/en.lproj/ClearPrivateDataConfirm.strings
XLIFF has target-language="en" for Shared/Supporting Files/en.lproj/BookmarkPanel.strings
XLIFF has target-language="en" for Shared/en.lproj/ClearHistoryConfirm.strings
XLIFF has target-language="en" for Shared/en.lproj/Intro.strings
XLIFF has target-language="en" for Shared/en.lproj/Search.strings
XLIFF has target-language="en" for Shared/en.lproj/3DTouchActions.strings
XLIFF has target-language="en" for Shared/en.lproj/Storage.strings
XLIFF has target-language="en" for Shared/en.lproj/AuthenticationManager.strings
XLIFF has target-language="en" for Shared/en.lproj/Today.strings
XLIFF has target-language="en" for WidgetKit/en.lproj/Localizable.strings
XLIFF has target-language="en" for Shared/en.lproj/PrivateBrowsing.strings
XLIFF has target-language="en" for WidgetKit/Base.lproj/QuickActionIntent.intentdefinition
XLIFF has target-language="en" for Shared/Supporting Files/en.lproj/BookmarkPanelDeleteConfirm.strings
XLIFF has target-language="en" for Client/en.lproj/InfoPlist.strings
XLIFF has target-language="en" for Shared/en.lproj/FindInPage.strings
XLIFF has target-language="en" for Shared/en.lproj/ErrorPages.strings
XLIFF has target-language="en" for Shared/en.lproj/Default Browser.strings
XLIFF has target-language="en" for Shared/en.lproj/Menu.strings
XLIFF has target-language="en" for Shared/en.lproj/Localizable.strings
XLIFF has target-language="en" for Shared/en.lproj/Shared.strings
XLIFF has target-language="en" for Shared/Supporting Files/en.lproj/ToolbarLocation.strings
	Please use the same locale codes for the contents.json, XLIFF, and .lproj directory names
```

I wonder if exporting the en-US locale explicitly will fix your issues with English content. 
Hopefully, that should also remove the need to manually remove strings from .lproj files, like I had to do recently (strings were still exported).

Also wonder if the en.lproj files are actually needed at this point.

**Comment by flodolo** [comment]
If I run `swift run LocalizationTools --export --project-path ../firefox-ios/Client.xcodeproj --l10n-project-path ../firefoxios-l10n --locale en-US`, this fixes the `target-language` in `en-US/firefox-ios.xliff`.

Then, if I run `swift run LocalizationTools --import --project-path ../firefox-ios/Client.xcodeproj --l10n-project-path ../firefoxios-l10n --locale en-US`, I don't have errors anymore. The diff is not as large, because en-US is not actually translated, but that could be easily solved when exporting the strings in the l10n repo.

**Comment by flodolo** [comment]
[This](https://github.com/mozilla-mobile/LocalizationTools/blob/main/Sources/LocalizationTools/tasks/ExportTask.swift#L182-L186) is the reason why both `en` and `en-US` output in `en-US/firefox-ios.xliff`

File: Sources/LocalizationTools/main.swift (hunk 1 above)
Comment (electricRGB) [reply Sources/LocalizationTools/main.swift:49]: I haven't worked much with the FileManager, but this seems correct, in general.

**Review by electricRGB — APPROVED** [review]

**Comment by flodolo** [comment]
Thanks. Could you merge this, since I don't have any right on the repo?

Side question: do you want me to make the change for `en-US` instead of `en` in the XLIFF file, so that next import will create the `en-US.lproj` folders?

**Comment by electricRGB** [comment]
Heya, let's make that change too, yeah, that'd be great. Should this go in this PR?

**Comment by flodolo** [comment]
> Heya, let's make that change too, yeah, that'd be great. Should this go in this PR?

No, this PR is complete. The change for en-US would go into the l10n repository.

**Pull request merged**